plugins/presentation/src/components/internal/Table/TableControls.svelte

```diff
@@ -0,0 +1,93 @@
+<script lang="ts">
+  import { createEventDispatcher, getContext } from 'svelte'
+  import { filter } from './utils'
+  const dispatch = createEventDispatcher()
+  const stateContext = getContext('table-state')
+
+  export let index = -1
+  export let text = ''
+
+  export let labels = {
+    placeholder: 'Search'
+  }
+
+  function onSearch (event) {
+    const state = stateContext.getState()
+    const detail = {
+      originalEvent: event,
+      filter,
+      index,
+      text,
+      page: state.page,
+      pageIndex: state.pageIndex,
+      pageSize: state.pageSize,
+      rows: state.filteredRows
+    }
+    dispatch('search', detail)
+
+    if (detail.preventDefault !== true) {
+      if (detail.text.length === 0) {
+        stateContext.setRows(state.rows)
+      } else {
+        stateContext.setRows(detail.rows.filter((r) => detail.filter(r, detail.text, index)))
+      }
+      stateContext.setPage(0, 0)
+    } else {
+      stateContext.setRows(detail.rows)
+    }
+  }
+</script>
+
+<div class="control-container">
+  <div class="search">
+    <input
+      type="search"
+      title={labels.placeholder}
+      placeholder={labels.placeholder}
+      bind:value={text}
+      on:keydown={onSearch} />
```

Comment:
Looks like search string is updated with previous value. For example, if type some text and delete it, search results will not be updated